### PR TITLE
Upgrade Terraform Plugin SDK

### PR DESCRIPTION
**Pull request by szh**
### Desired Outcome

Dependabot is flagging github.com/hashicorp/go-getter for CVE-2022-29810. go-getter is an indirect reference, brought in through github.com/hashicorp/terraform-plugin-sdk. We currently use v1.17.0 of this SDK, but the 1.x line is no longer updated. We need to move to the v2 line (which removes the dependency on go-getter entirely as far as I can see.)

Looking at the [v2 Upgrade Guide](https://www.terraform.io/plugin/sdkv2/guides/v2-upgrade-guide), this will be more than just a version bump. We need to rework the provider as the terraform.ResourceProvider interface has been removed. There are also additional usages of Context and new features we may want to use as well as other removed functions (though I didn't see signs we used any of those functions). Upgrading will also mean we drop support for Terraform 0.11. HashiCorp suggests a new major version to reflect dropping this support.

### Implemented Changes

- Updated go modules with new SDK and it's dependencies
- Updated import paths
- Removed use of `terraform.ResourceProvider` interface and use concrete type instead.

### Connected Issue/Story

CyberArk internal issue link: [ONYX-22156](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-22156)

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [x] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

**Review by gl-johnson — COMMENTED** [review]
Left a comment.

Another question - does the MPLv2 section of NOTICES.txt need to be updated with the new terraform SDK version?

File: go.mod [hunk 1]
```diff
@@ -1,76 +1,44 @@
...
+	github.com/hashicorp/go-hclog v0.15.0 // indirect
 	github.com/hashicorp/go-multierror v1.0.0 // indirect
-	github.com/hashicorp/go-plugin v1.3.0 // indirect
-	github.com/hashicorp/go-safetemp v1.0.0 // indirect
+	github.com/hashicorp/go-plugin v1.4.0 // indirect
 	github.com/hashicorp/go-uuid v1.0.1 // indirect
 	github.com/hashicorp/go-version v1.3.0 // indirect
 	github.com/hashicorp/hcl/v2 v2.8.2 // indirect
-	github.com/hashicorp/terraform-svchost v0.0.0-20200729002733-f050f53b9734 // indirect
+	github.com/hashicorp/terraform-plugin-sdk/v2 v2.4.3
```
[the remark this answers is not in the document]
Comment (szh) [reply go.mod]: Good catch. Fixed that up

**Comment by szh** [comment]
IIRC @andytinkham said we should ignore the Snyk license tests for now.

**Review by gl-johnson — APPROVED** [review]
> IIRC @andytinkham said we should ignore the Snyk license tests for now.

In that case, LGTM!

**Pull request merged**